Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -227,6 +260,48 @@ namespace AZ::Dom
         return m_entries == other.m_entries;
     }
 
+    bool Path::operator<(const Path& other) const
+    {
+        auto ComparePathEntry = [](const PathEntry& lhs, const PathEntry& rhs)
+        {
+            return lhs == rhs;
+        };
+        // Find the PathEntry where the two Path objects differ, if any
+        auto [thisPathIter, otherPathIter] = AZStd::mismatch(
+            GetEntries().begin(), GetEntries().end(),
+            other.GetEntries().begin(), other.GetEntries().end(),
+            ComparePathEntry);
+
+        bool isLessThan = false;
+        // If the paths differ then compare the differing PathEntries
+        if (thisPathIter != GetEntries().end() && otherPathIter != other.GetEntries().end())
+        {
+            isLessThan = *thisPathIter < *otherPathIter;
+        }
+        // If one Path is a prefix of the other, or the paths are identical, then the shorter
+        // Path is the smaller Path
+        else
+        {
+            isLessThan = other.GetEntries().size() > GetEntries().size();
+        }
+        return isLessThan;
+    }
+
+    bool Path::operator>(const Path& other) const
+    {
+        return other.operator<(*this);
+    }
+
+    bool Path::operator<=(const Path& other) const
+    {
+        return operator<(other) || operator==(other);
```

Comment:
This shouldn't use `operator==` here at all.
```suggestion
        return !other.operator<(*this);
```

Reply:
Nice, that way everything is using the operator<? Done.